multiraft/multiraft.go

```diff
@@ -651,6 +651,33 @@ func (s *state) addNode(nodeID proto.RaftNodeID, groupIDs ...proto.RangeID) erro
 	return nil
 }
 
+// removeNode removes a node from a group.
+func (s *state) removeNode(nodeID proto.RaftNodeID, groupID proto.RangeID) error {
+	node, ok := s.nodes[nodeID]
+	if !ok {
+		return util.Errorf("cannot remove unknown node %s", nodeID)
+	}
+	g := s.groups[groupID]
```

Comment:
looks like there's a pointer to the group in the caller - maybe pass it in instead of the id?